tests/php/classes/test-class-wc-connect-compatibility-wcshipping-packages.php

```diff
@@ -0,0 +1,284 @@
+		$this->set_has_completed_migration( false );
+
+		WC_Connect_Compatibility_WCShipping_Packages::maybe_enable();
+
+		$this->assert_is_rest_controller_registration_hook_registered( true );
+		$this->assert_are_option_interception_hooks_registered( false );
+	}
+
+	public function test_it_does_not_enable_anything_if_wcshipping_is_not_active() {
+		$this->set_is_wcshipping_active( false );
+		$this->set_has_completed_migration( true );
+
+		WC_Connect_Compatibility_WCShipping_Packages::maybe_enable();
+
+		$this->assert_is_rest_controller_registration_hook_registered( false );
+		$this->assert_are_option_interception_hooks_registered( false );
+	}
+
+	public function test_wc_connect_options_contains_wcshipping_packages_mapped_to_wcservices_format() {
+		$this->set_is_wcshipping_active( true );
+		$this->set_has_completed_migration( true );
+
+		WC_Connect_Compatibility_WCShipping_Packages::maybe_enable();
+
+		/*
+		 * self::EXAMPLE_WCSHIPPING_OPTIONS["packages"] mapped to WCS&T's format and unused keys removed.
+		 *
+		 * Note the expected result for the item with leftover fields uses the following approach:
+		 * - if both a WCShipping key and a WCS&T key is present (perhaps as a result of options migration),
+		 *   prefer the WCShipping key and disregard the WCS&T one (e.g. will use "boxWeight");
+		 * - for determining package type (box/envelope), if the "type" (current WCShipping approach), "isLetter"
+		 *   (deprecated WCShipping approach), and "is_letter" (WCS&T approach) keys are present, it will use the "type"
+		 *   key and disregard the rest, even if they are conflicting (hypothetical situation);
+		 * - keys that aren't contained in the "KEYS_USED_BY_*" const will be removed.
+		 */
+		$expected_custom_wcshipping_packages_mapped_to_wcservices_format = array(
+			array(
+				'box_weight'       => 301,
+				'inner_dimensions' => '302 x 303 x 304',
+				'is_letter'        => false,
+				'is_user_defined'  => true,
+				'max_weight'       => 306,
+				'name'             => 'WCShipping custom box',
+			),
+			array(
+				'box_weight'       => 401,
+				'inner_dimensions' => '402 x 403 x 404',
+				'is_letter'        => true,
+				'is_user_defined'  => true,
+				'max_weight'       => 406,
+				'name'             => 'WCShipping custom envelope',
+			),
+			array(
+				'box_weight'       => 501,
+				'inner_dimensions' => '503 x 504 x 505',
+				'is_letter'        => false,
+				'is_user_defined'  => true,
+				'max_weight'       => 510,
+				'name'             => 'WCShipping custom envelope with some leftover fields from WCS&T migration or previous package data formatting',
+			),
+		);
+
+		$this->assertEquals(
+			array_merge(
+				self::EXAMPLE_WC_CONNECT_OPTIONS,
+				array(
+					'packages'            => $expected_custom_wcshipping_packages_mapped_to_wcservices_format,
+					'predefined_packages' => self::EXAMPLE_WCSHIPPING_OPTIONS['predefined_packages'],
+				)
+			),
+			get_option( 'wc_connect_options' )
```

Comment:
Nice test, the setup starts with something in both `wc_connect_options` and `wcshipping_options`. This merges the two together when `wc_connect_options` is requested. 👍